File: dkb/convert_dkb_date.go
package dkb

import "time"

func convertDkbDate(input string) (string, error) {
	inputFormat := "02.01.06"
	outputFormat := "2006-01-02"

	parsedDate, inputFormatErr := time.Parse(inputFormat, input)
	if inputFormatErr != nil {
		return "", inputFormatErr
	}

	return parsedDate.Format(outputFormat), nil
}

File: dkb/convertDkbDate.ts
export function convertDkbDate(input: string): string {
  const [day, month, year] = input.split(".");
  
  if (!day || !month || !year || 
      isNaN(parseInt(day)) || 
      isNaN(parseInt(month)) || 
      isNaN(parseInt(year))) {
    throw new Error(`Invalid date format: ${input}`);
  }

  const fullYear = parseInt(year) + 2000;
  
  const paddedDay = day.padStart(2, "0");
  const paddedMonth = month.padStart(2, "0");
  
  return `${fullYear}-${paddedMonth}-${paddedDay}`;
}
